pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -187,48 +186,36 @@ def stop_at(obj):
             # If next gate is of the same type, we can merge the angles
             if current_gate.name == next_gate.name and current_gate.wires == next_gate.wires:
```

Comment:
Could we use `isinstance` instead of `name`?

Reply:
I suppose there is always an edge case where we would want merging and it would be missed?
- For the `.name`-based logic, equivalent gates with differing names are not hit.
- For `isinstance`-based logic, `current_gate` being equivalent to `next_gate` but `next_gate` being of an inheriting type would not be hit.


I believe the `isinstance`-based logic is a tad better in that sence :)

Reply:
I only had in mind that `isinstance` is a bit faster than comparing the `name` strings :)